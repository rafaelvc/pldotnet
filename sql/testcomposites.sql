
CREATE TYPE NameAge AS (
    name            text,
    age             integer
);

CREATE OR REPLACE FUNCTION hello(a name_age) RETURNS text AS $$
int n = 1;
FormattableString res;
a.age += n;
res = $"Hello M(r)(s). ${a.name}! Your age plus ${n} is equal to ${a.age}."
return res.ToString();
$$ LANGUAGE plcsharp;
SELECT hello(('Rafael Cabral', 38));

--CREATE OR REPLACE FUNCTION hello_aaa(a NameAge) RETURNS text AS $$
--BEGIN
--    RETURN a.name;
--END;
--$$ LANGUAGE plpgsql;
--SELECT hello_aaa(('Rafael Cabral', 38));



