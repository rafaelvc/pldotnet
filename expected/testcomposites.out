CREATE TYPE Person AS (
    name            text,
    age             integer,
    weight          double precision,
    height          real,
    salary          double precision,
    married         boolean
);
CREATE OR REPLACE FUNCTION helloPersonAge(per Person) RETURNS integer AS $$
    FormattableString res;
    res = $"Hello M(r)(s). {per.name}! Your age {per.age}, {per.weight}, {per.height}, {per.salary}. {per.married}";
    return per.age;
$$ LANGUAGE plcsharp;
SELECT helloPersonAge(('John Smith', 38, 85.5, 1.71, 999.999, true)) = integer '38';
 ?column? 
----------
 t
(1 row)

CREATE OR REPLACE FUNCTION helloPerson(p Person) RETURNS Person AS $$
    int n = 1;
    FormattableString res;
    p.age += n;
    res = $"Hello M(r)(s). {p.name}! Your age {p.age}, {p.weight}, {p.height}, {p.salary}. {p.married}";
    return p;
$$ LANGUAGE plcsharp;
SELECT helloPerson(('John Smith', 38, 85.5, 1.71, 999.999, true));
              helloperson              
---------------------------------------
 ("John Smith",39,85.5,1.71,999.999,t)
(1 row)

